numba_cuda/numba/cuda/codegen.py

```diff
@@ -280,14 +283,17 @@ def get_cubin(self, cc=None):
             # `-ptx` flag is meant to view the optimized PTX for LTO objects.
             # Non-LTO objects are not passed to linker.
             self._link_all(linker, cc, ignore_nonlto=True)
+            ptx = linker.get_linked_ptx()
 
-            ptx = linker.get_linked_ptx().decode("utf-8")
+            if config.CUDA_USE_NVIDIA_BINDING:
+                ptx = ptx.code
+            ptx = ptx.decode("utf-8")
 
             print(("ASSEMBLY (AFTER LTO) %s" % self._name).center(80, "-"))
             print(ptx)
             print("=" * 80)
 
-        linker = driver.Linker.new(
+        linker = driver._Linker.new(
             max_registers=self._max_registers, cc=cc, lto=self._lto
         )
         self._link_all(linker, cc, ignore_nonlto=False)
```

Comment:
I can't comment on line 294 / 300 just below as it's not part of the diff, but I suspect if it was changed so that `cubin = linker.complete().code` if the NVIDIA binding is in use, then the other changes to use `<variable>.code` (https://github.com/NVIDIA/numba-cuda/pull/133/files#diff-bf7c3a638421914e7463c6a23bd6293b740ed0c3e22267a7469091f70a9ad746R26, https://github.com/NVIDIA/numba-cuda/pull/133/files#diff-626c1086aae9de03cd89533ef1e1b9f7b1279e834b54aab10ab3c271739def53R1690) wouldn't be needed and the interface would be more uniform across the linkers. (Not a necessary change as a lot of this stuff will churn anyway as we move more to cuda-core / cuda-bindings, but noticed the use of `.code` in other places).